extensions/typescript-language-features/src/utils/configuration.ts

```diff
@@ -146,4 +149,12 @@ export class TypeScriptServiceConfiguration {
 	private static readUseSeparateSyntaxServer(configuration: vscode.WorkspaceConfiguration): boolean {
 		return configuration.get<boolean>('typescript.tsserver.useSeparateSyntaxServer', true);
 	}
+
+	private static readMaxTsServerMemory(configuration: vscode.WorkspaceConfiguration): number {
+		const memoryInMB = configuration.get<number>('typescript.tsserver.maxTsServerMemory', 0);
+		if (!Number.isSafeInteger(memoryInMB) || memoryInMB < 0) {
+			return 0;
```

Comment:
We probably should add a lower bound here, such as 128mb. We do this for a few other settings like fontSize